src/device-registry/utils/create-event.js

```diff
@@ -1089,29 +1090,33 @@ const createEvent = {
             success: true,
             message: "utilising cache...",
             data: resultJSON,
+            status: httpStatus.OK,
           });
         } else if (err) {
           logger.error(`unable to get cache --- ${JSON.stringify(err)}`);
           callback({
             success: false,
             message: "Internal Server Error",
             errors: { message: err.message },
+            status: httpStatus.INTERNAL_SERVER_ERROR,
           });
         } else {
           callback({
             success: false,
             message: "no cache present",
             errors: { message: "no cache present" },
+            status: httpStatus.INTERNAL_SERVER_ERROR,
           });
         }
       });
     } catch (error) {
       logger.error(`internal server error -- ${error.message}`);
-      return {
+      callback({
```

Comment:
@BenjaminSsempala , this was causing the timeout issue. The function utilises a callback function as an input and yet we were just returning fwwaaaa, like other normal parameter funcs....  cc: @NoahNsimbe